workbench-core/datasets/src/models/dataSetPermission.ts

```diff
@@ -3,13 +3,17 @@
  *  SPDX-License-Identifier: Apache-2.0
  */
 
-import { DataSetsAccessLevel } from './dataSetsAccessLevel';
+import { z } from 'zod';
+import { DataSetsAccessLevelParser } from './dataSetsAccessLevel';
 
-export interface DataSetPermission {
+// eslint-disable-next-line @rushstack/typedef-var
+export const DataSetPermissionParser = z.object({
   /** the user or group associated with the access level */
-  identity: string;
+  identity: z.string(),
   /** the type of subject (group or user) */
-  identityType: string;
+  identityType: z.string(),
```

Comment:
```suggestion
  identityType: z.union([z.literal('GROUP'), z.literal('USER')]);,
```

Reply:
Intentionally avoided at this level.

I leave it up to the implementation of the authorization plugin to choose the identity types appropriate for the authz backend and the specific use case. Those are correct for our reference implementation, but defining that union in at this interface level limits future implementations.